dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/supplier/UserSupplier.java

```diff
@@ -55,16 +58,34 @@
     @NonNull
     private final IdentifiableObjectManager manager;
 
+    @NonNull
+    private final UserService userService;
+
     @Override
     public void preheatAdd( TrackerImportParams params, TrackerPreheat preheat )
     {
         Set<String> userUids = params.getEvents().stream()
             .filter( Objects::nonNull )
             .map( Event::getAssignedUser )
+            .filter( Objects::nonNull )
+            .map( UserInfo::getUid )
             .filter( CodeGenerator::isValidUid )
             .collect( Collectors.toSet() );
 
-        preheat.put( TrackerIdentifier.UID,
+        Set<String> usernames = params.getEvents().stream()
+            .filter( Objects::nonNull )
+            .map( Event::getAssignedUser )
+            .filter( Objects::nonNull )
+            .map( UserInfo::getUsername )
+            .collect( Collectors.toSet() );
+
+        List<User> users = userService.getUsersByUsernames( usernames );
+
+        Set<User> validUsers = new HashSet<>( DetachUtils.detach( UserMapper.INSTANCE, users ) );
```

Comment:
For my understanding since I just worked on a Supplier. Is it necessary to detach?

Reply:
Detach is necessary to load all the lazy fields. Maybe in this case is not really necessary as we are using just simple fields but I would rather use it for consistency and to be safe if in the future we are going to change the mapper and use more fields from User